cpp_utils/BLEDevice.cpp

```diff
@@ -162,10 +162,22 @@ gatts_event_handler BLEDevice::m_customGattsHandler = nullptr;
 			break;
 	} // switch
 	for(auto &myPair : BLEDevice::getPeerDevices(true)) {
-		conn_status_t conn_status = (conn_status_t)myPair.second;
-		if(((BLEClient*)conn_status.peer_device)->getGattcIf() == gattc_if || ((BLEClient*)conn_status.peer_device)->getGattcIf() == ESP_GATT_IF_NONE || gattc_if == ESP_GATT_IF_NONE){
-			((BLEClient*)conn_status.peer_device)->gattClientEventHandler(event, gattc_if, param);
+		BLEClient* client = (BLEClient*)((conn_status_t)myPair.second).peer_device;
```

Comment:
Hi, could you instead of change BLEDevice.cpp here change code in BLEClient.cpp:
```
		case ESP_GATTC_REG_EVT: {
			if(m_appId == evtParam->reg.app_id){
				ESP_LOGI(__func__, "register app id: %d, %d, gattc_if: %d", m_appId, evtParam->reg.app_id, gattc_if);
				m_gattc_if = gattc_if;
				m_appId = evtParam->reg.app_id;
				m_semaphoreRegEvt.give();
			}
			break;
		} // ESP_GATTC_REG_EVT
```